### PR TITLE
support jq for mac in bigquery_test

### PR DESCRIPTION
**Pull request by BenTheElder**
This works but it's not necessarily ideal and test-infra Bazel usage on macOS still really only works from Planter for now :man_shrugging:

### CODE REVIEW

File: metrics/BUILD [hunk 1]
```diff
@@ -13,17 +13,38 @@ filegroup(
     tags = ["automanaged"],
 )
 
+config_setting(
```
Comment (ixdy) [inline metrics/BUILD]: is it possible to put these in the `WORKSPACE` instead and then use the correct URL depending on the config setting?

Comment (BenTheElder) [reply metrics/BUILD]: These suggest not but I'll verify:
https://github.com/bazelbuild/bazel/issues/1482
https://github.com/bazelbuild/bazel/issues/1485

**Comment by BenTheElder** [comment]
fixes https://github.com/kubernetes/test-infra/issues/4850

File: metrics/BUILD (hunk 1 above)
Comment (BenTheElder) [reply metrics/BUILD]: Yeah I get an error containing `config_setting cannot be in the WORKSPACE file`
At least the jq OSX binary is only 635kb.

Comment (ixdy) [reply metrics/BUILD]: yeah, it looks like the only way to do this is to create a `repository_rule`, which is unfortunate.

that's probably generalizable and reusable, though I don't know if anyone has done that yet - seems like everyone rolls their own.

File: metrics/BUILD [hunk 2]
```diff
@@ -13,17 +13,38 @@ filegroup(
...
-        ":all-srcs",
-        "@jq//file",
-    ],
+    args = select({
+        ":k8": ["--jq=$(location @jq_linux//file)"],
+        ":darwin": ["--jq=$(location @jq_osx//file)"],
+    }),
+    data = select({
+        ":k8": [
+            ":all-srcs",
```
Comment (ixdy) [inline metrics/BUILD]: factor `":all-srcs"` out of the `select`.

also, this should really be `glob(["configs/*"])` rather than `":all-srcs"`.

Comment (BenTheElder) [reply metrics/BUILD]: done, had to also include `README.md`. I'm not sure if the way I wrote it is idiomatic or not.

**Comment by BenTheElder** [comment]
agh travis! reminder to finish #1342

**Pull request merged**